cpp/src/io/reader.cpp

```diff
@@ -247,21 +253,29 @@ las::Points Reader::GetPointsWithinBox(const Box &box, double min_resolution)
     return out;
 }
 
-int32_t Reader::GetDepthAtResolution(double resolution) const
+int32_t Reader::GetDepthAtResolution(double resolution)
 {
+    // If query resolution is <=0 return the octree's max depth
     if (resolution <= 0)
-        throw std::runtime_error("Reader::GetDepthAtResolution: Query resolution must be greater than 0.");
+    {
+        int32_t max_depth = 0;
+        // Get all nodes in octree
+        for (const auto &node : GetAllChildren())
+        {
+            if (node.key.d > max_depth)
+                max_depth++;
```

Comment:
```suggestion
                max_depth = node.key.d;
```
or: `max_depth = std::min(node.key.d, max_depth);